### PR TITLE
Adding Django 1.10 Fix

### PR DESCRIPTION
**Pull request by bentleycook**
## Purpose
Django 1.10 changed the way Middleware functions. Using raygun4py 3.1.4 with Python 3 and Django 1.10 results in a TypeError (same problem as documented in [this](http://stackoverflow.com/questions/38792686/installation-of-django-debug-toolbar-typeerror-init-takes-1-positional-a) SO question).

Following the Django docs [here](https://docs.djangoproject.com/en/1.10/topics/http/middleware/#upgrading-middleware), I used the deprecation MiddlewareMixin and a pattern similar to the one used in the fix for django-debug-toolbar. This resolves the `TypeError` but results in `RecursionError`.

I'll keep digging around but wanted to toss this half-finished fix in a PR in case someone else was looking at the same problem.

## Related Findings
Looks like v3.1.0 added support for Py3 here: https://github.com/MindscapeHQ/raygun4py/pull/37. Couldn't find any mention of Django 1.10
Similar Django Debug Toolbar issue: https://github.com/jazzband/django-debug-toolbar/issues/853

### CODE REVIEW

**Comment by fundead** [comment]
Thanks for the discovery and triage of this issue, and the WIP PR - much appreciated. Just wondering if you had any more details on the RecursionError - were you seeing this upon import/config of the provider, or upon sending an error for instance from a django view?

**Comment by bentleycook** [comment]
Sorry, haven't had a chance to get back to this PR yet. I was seeing the
recursion error occur when trying to send an error to raygun. It was
happening when trying to pickle the error data for raygun.

File: python3/raygun4py/middleware/django.py [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+try:
+    from django.utils.deprecation import MiddlewareMixin
+except ImportError:
+    MiddlewareMixin = object
 
-class Provider(object):
+class Provider(MiddlewareMixin):
+
+    def __init__(self, get_response=None):
+        super().__init__(get_response)
```
Comment (donspaulding) [inline python3/raygun4py/middleware/django.py]: If you were to change this line to `self.get_response = get_response` I think the `RecursionError` would go away and the middleware would work.

(I'm not using this client library at the moment, but I'm upgrading a bunch of legacy Django middleware and came across this one and the problem intrigued me.  Feel free to ignore this if it's not helpful).

**Comment by ewen** [comment]
Having Django's `django.utils.deprecation.MiddlewareMixin` inherit from the Raygun provider worked for me. This is the class I'm using in our project:

```
from raygun4py.middleware.django import Provider
from django.utils.deprecation import MiddlewareMixin


class RayGunMiddleware(MiddlewareMixin, Provider):
    pass
```

**Pull request merged**